packages/ia-components/sandbox/theatres/with-youtube-spotify/audio-with-youtube-spotify-main.jsx

```diff
@@ -212,13 +221,14 @@ class AudioPlayerWithYoutubeSpotify extends Component {
       tracklistToShow, trackSelected, channelToPlay, albumData
     } = this.state;
     const {
-      title, itemPhoto, playSamples, externalSources = [], identifier, collection
+      title, itemPhoto, playSamples, externalSources = [], identifier, collection, externalSourcesDisplayValues, creator
     } = albumData;
     let audioPlayerChannelLabel;
-    if (channelToPlay === 'archive') {
-      audioPlayerChannelLabel = `Archive${playSamples ? ' Samples' : ''}`;
+    const isArchiveChannel = channelToPlay === 'archive';
+    if (isArchiveChannel) {
+      audioPlayerChannelLabel = `${playSamples ? ' Samples' : 'Internet Archive'}`;
     } else {
-      audioPlayerChannelLabel = channelToPlay;
+      audioPlayerChannelLabel = externalSourcesDisplayValues[channelToPlay] || '';
```

Comment:
Where is `channelToPlay` defined -- is it possible to be undefined?

Reply:
good catch, it's has a default value, won't be undefined